misc/distributed_search/outer.sh

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env bash
+
+#SBATCH --job-name=dist_AD
+#SBATCH --time=21-00:00:00
+#SBATCH --output=./logs/%j-%x-slurm.out
+
+# slurm parameters
+nnodes=1
+ntasks_per_node=1
+cpus=32
+mem='250G'
+# Search parameters
+input_directory="/fs/home/brennsteiner/alphadia/distributed_search_test/"
+input_filename="file_list.csv"
+target_directory="/fs/home/brennsteiner/alphadia/distributed_search_test/search/"
+library_path="/fs/home/brennsteiner/alphadia/distributed_search_test/48_fraction_hela_PaSk_orbitrap_ms2.hdf"
+fasta_path="/fs/home/brennsteiner/alphadia/distributed_search_test/uniprotkb_proteome_UP000005640_2023_11_12.fasta"
+first_search_config_filename="config.yaml"
+second_search_config_filename="config.yaml"
+# Search flags
+predict_library=1
+first_search=1
+mbr_library=1
+second_search=1
+lfq=1
+
+while [[ "$#" -gt 0 ]]; do
+	case $1 in
+		# Search parameters
+		--input_directory) input_directory="$2"; shift ;;
+		--input_filename) input_filename="$2"; shift ;;
+		--target_directory) target_directory="$2"; shift ;;
+		--library_path) library_path="$2"; shift ;;
+		--fasta_path) fasta_path="$2"; shift ;;
+		--first_search_config_filename) first_search_config_filename="$2"; shift ;;
+		--second_search_config_filename) second_search_config_filename="$2"; shift ;;
+		# SLURM parameters
+		--nnodes) nnodes="$2"; shift ;;
+		--ntasks_per_node) ntasks_per_node="$2"; shift ;;
+		--cpus) cpus="$2"; shift ;;
+		--mem) mem="$2"; shift ;;
+		# Search flags: if a flag is present, set to true
+		--predict_library) predict_library="$2"; shift ;;
+		--first_search) first_search="$2"; shift ;;
+		--mbr_library) mbr_library="$2"; shift ;;
+		--second_search) second_search="$2"; shift ;;
+		--lfq) lfq="$2"; shift ;;
+		*) echo "Unknown parameter $1"; shift ;;
+	esac
+	shift
+done
+
+	# create logs directory if it does not exist
+	mkdir -p ./logs
+
+	# remove and create target directory
+	rm -rf ${target_directory}
+	mkdir ${target_directory}
+
+	predicted_library_directory="${target_directory}predicted_speclib/"
```

Comment:
You mean e.g. in order to restart the search later and use an existing speclib that was made from the previous search? In that case it might even be better to use a timestamp and make the folder unique, rather than having to know which speclibs already exist or not?

Reply:
I was thinking just a fixed number to have the consecutive steps ordered nicely in the file system

Reply:
ah, got you! 👍